src/parsing/Parse_rule.ml

```diff
@@ -453,6 +453,31 @@ let parse_python_expression env key s =
 
 let parse_metavar_cond env key s = parse_python_expression env key s
 
+let rec parse_type env key (str, tok) =
+  match env.languages.target_analyzer with
+  | Xlang.L (lang, _) ->
+      let str = wrap_type_expr env key lang str in
+      try_and_raise_invalid_pattern_if_error env (str, tok) (fun () ->
+          Parse_pattern.parse_pattern lang ~print_errors:false str)
+      |> unwrap_typed_metavar env key
+  | Xlang.LRegex
+  | Xlang.LSpacegrep
+  | Xlang.LAliengrep ->
+      error_at_key env.id key
+        "`type` is not supported with regex, spacegrep or aliengrep."
+
+and wrap_type_expr env key lang str =
+  match lang with
+  | Lang.Java -> spf "(%s x)" str
```

Comment:
what is this 'x'?
Maybe add a comment like  (* coupling: we use 'x' here because it's a hack we use in XXX.ml to parse YYY *)

Reply:
Or just explain that we don't care about this 'x' in unwrap_typed_metavar.